File: src/main/java/net/cjsah/console/command/argument/GroupArgument.java
package net.cjsah.console.command.argument;

import net.cjsah.console.Console;
import net.cjsah.console.command.StringReader;
import net.cjsah.console.command.context.CommandContext;
import net.cjsah.console.exceptions.BuiltExceptions;
import net.cjsah.console.exceptions.CommandException;
import net.mamoe.mirai.contact.Friend;
import net.mamoe.mirai.contact.Group;

public class GroupArgument implements Argument<Group> {

    private GroupArgument() {
    }

    private static GroupArgument group() {
        return new GroupArgument();
    }

    private static Group getGroup(CommandContext context, String name) {
        return context.getArgument(name, Group.class);
    }

    @Override
    public Group parse(StringReader reader) throws CommandException {
        int start = reader.getCursor();
        long id = reader.readLong();
        Group group = Console.INSTANCE.getBot().getGroup(id);
        if (group == null) {
            reader.setCursor(start);
            throw BuiltExceptions.groupNotFound().createWithContext(reader, id);
        }
        return group;
    }
}

File: src/main/kotlin/net/cjsah/console/command/argument/GroupArgument.kt
@file:Suppress("unused")

package net.cjsah.console.command.argument

import net.cjsah.console.Console.getBot
import net.cjsah.console.command.StringReader
import net.cjsah.console.command.context.CommandContext
import net.cjsah.console.exceptions.BuiltExceptions
import net.mamoe.mirai.contact.Group

class GroupArgument private constructor() : Argument<Group> {

    companion object {
        @JvmStatic
        fun group() = GroupArgument()

        @JvmStatic
        fun getGroup(context: CommandContext, name: String) = context.getArgument(name, Group::class.java)
    }

    override fun parse(reader: StringReader): Group {
        val start = reader.getCursor()
        val id = reader.readLong()
        val group = getBot().getGroup(id)
        if (group == null) {
            reader.setCursor(start)
            throw BuiltExceptions.GROUP_NOT_FOUND.createWithContext(reader, id)
        }
        return group
    }
}